app/components/goodcity/code-search-overlay.js

```diff
@@ -22,6 +23,18 @@ export default Ember.Component.extend(SearchMixin, {
     this._super("code-search-overlay");
   },
 
+  async didRender() {
+    const hasData = this.get("store")
+      .peekAll("code")
+      .get("length");
+
+    if (this.get("open") && !hasData) {
+      await this.runTask(async () => {
+        await this.get("store").query("code", { stock: true });
+      }, ERROR_STRATEGIES.MODAL);
+    }
+  },
```

Comment:
this kinda feels like the preload mechanism isn't doing its role properly. Also making requests in the background with no subsequent handling of the returned data feels a bit off